crates/sargon/src/system/sargon_os/sargon_os_personas.rs

```diff
@@ -0,0 +1,1207 @@
+    }
+
+    /// Creates many new non securified personas **WITHOUT** add them to Profile, using the *main* "Babylon"
+    /// `DeviceFactorSource` and the "next" indices for this FactorSource as derivation paths.
+    ///
+    /// If you want to add them to Profile, call `add_personas(personas)`
+    ///
+    /// # Emits Event
+    /// Emits `Event::FactorSourceUpdated { id: FactorSourceID }` since the date in
+    /// `factor_source.common.last_used` is updated.
+    pub async fn batch_create_unsaved_personas_with_bdfs(
+        &self,
+        network_id: NetworkID,
+        count: u16,
+        name_prefix: String,
+    ) -> Result<(Personas, InstancesInCacheConsumer)> {
+        let bdfs = self.bdfs()?;
+        self.batch_create_unsaved_personas_with_factor_source(
+            bdfs.into(),
+            network_id,
+            count,
+            name_prefix,
+        )
+        .await
+    }
+
+    /// Creates many new non securified personas **WITHOUT** add them to Profile, using selected factor source
+    ///  and the "next" indices for this FactorSource as derivation paths.
+    ///
+    /// If you want to add them to Profile, call `add_personas(personas)`
+    ///
+    /// # Emits Event
+    /// Emits `Event::FactorSourceUpdated { id: FactorSourceID }` since the date in
+    /// `factor_source.common.last_used` is updated.
+    pub async fn batch_create_unsaved_personas_with_factor_source(
+        &self,
+        factor_source: FactorSource,
+        network_id: NetworkID,
+        count: u16,
+        name_prefix: String,
+    ) -> Result<(Personas, InstancesInCacheConsumer)> {
+        self.batch_create_unsaved_personas_with_factor_source_with_derivation_outcome(
+            factor_source,
+            network_id,
+            count,
+            name_prefix,
+        )
+        .await
+        .map(|(x, y, _)| (x, y))
+    }
+    pub async fn batch_create_unsaved_personas_with_factor_source_with_derivation_outcome(
+        &self,
+        factor_source: FactorSource,
+        network_id: NetworkID,
+        count: u16,
+        name_prefix: String,
+    ) -> Result<(
+        Personas,
+        InstancesInCacheConsumer,
+        FactorInstancesProviderOutcomeForFactor,
+    )> {
+        let key_derivation_interactors = self.keys_derivation_interactors();
+
+        let profile = self.profile()?;
+
+        let (
+            factor_source_id,
+            personas,
+            instances_in_cache_consumer,
+            derivation_outcome,
+        ) = profile
+            .create_unsaved_personas_with_factor_source_with_derivation_outcome(
+                factor_source,
+                network_id,
+                count,
+                Arc::new(self.clients.factor_instances_cache.clone()),
+                key_derivation_interactors,
+                |idx| {
+                    DisplayName::new(format!("{} {}", name_prefix, idx))
+                        .expect("Should not use a long name_prefix")
```

Comment:
this `new` doesn't validate the length is less than N, but actually return `Err` only if empty (which won't be the case given we have the `idx` minimum.

Reply:
FIXED